src/tool/hpcrun/fnbounds/fnbounds_client.c

```diff
@@ -455,7 +455,17 @@ hpcfnbounds_child(void* fds_vp) {
   // Do the clone, and pass the result back to our parent through the shared memory space
   fds->pid = auditor_exports->clone(hpcfnbounds_grandchild,
     &server_stack[SERVER_STACK_SIZE * 1024], SIGCHLD | CLONE_VM, fds_vp);
-  return 0;
+
+  // Glibc before 2.24 will reset the internal pthread state if we return from
+  // this function. Since we're sharing a memory space with the parent this
+  // wreaks havoc. So instead die by a signal to get out.
+  struct sigaction act;
+  act.sa_handler = SIG_DFL;
+  act.sa_flags = 0;
+  auditor_exports->sigaction(SIGUSR1, &act, NULL);
+
+  raise(SIGUSR1);
```

Comment:
This idiom doesn't work on NOTS. The raise returns and then abort terminates with a SEGV. This causes everything to die, including the application. 
- I don't know why the process that raised an exception on itself doesn't die.
- I don't know why the child's death takes down the parent.